superpowered/Superpowered.d.ts

```diff
@@ -1,22 +1,491 @@
+    /** Global Spatializer Reverb room size. >= 0 and <= 1. */
+    static reverbRoomSize: number;
+    /** Global Spatializer Reverb pre-delay in milliseconds. 0 to 500. */
+    static reverbPredelayMs: number;
+    /** Global Spatializer Reverb frequency of the low cut in Hz (-12 db point). Default: 0 (no low frequency cut). */
+    static reverbLowCutHz: number;
+
+    /**
+     * Processes the audio.
+     * It's never blocking for real-time usage. You can change all properties on any thread, concurrently with process().
+     * If process() returns with true, the contents of output are replaced with the audio output. If process() returns with false, the contents of output are not changed.
+     */
+    process: (
+        /** Pointer to floating point numbers. 32-bit left channel or interleaved stereo input. */
+        inputLeft: SuperpoweredMemoryPointer,
+        /** Pointer to floating point numbers. 32-bit right channel input. Can be null, inputLeft will be used in this case as interleaved stereo input. */
+        inputRight: SuperpoweredMemoryPointer,
+        /** Pointer to floating point numbers. 32-bit left channel or interleaved stereo output. */
+        outputLeft: SuperpoweredMemoryPointer,
+        /** Pointer to floating point numbers. 32-bit right channel output. Can be null, outputLeft will be used in this case as interleaved stereo output. */
+        outputRight: SuperpoweredMemoryPointer,
+        /** Number of frames to process. Valid between 64-8192. */
+        bufferSize: number,
+        /** If true, audio will be added to whatever content is in outputLeft or outputRight. */   
+        mixInPlace: boolean       
+    ) => boolean;
+    
+    /**
+     * Outputs the Global Spatializer Reverb. Always call it in the audio processing callback, regardless if the effect is enabled or not for smooth, audio-artifact free operation. Should be called after every Spatializer's process() method.
+     * It's never blocking for real-time usage. You can change all properties of the globalReverb on any thread, concurrently with process().
+     * If process() returns with true, the contents of output are replaced with the audio output. If process() returns with false, the contents of output are not changed.
+     */
+    static reverbProcess: (
+        /** Pointer to floating point numbers. 32-bit interleaved stereo output. */
+        output: SuperpoweredMemoryPointer,
+        /** Number of framesto process. Should not be higher than 8192. */
+        bufferSize: number
+    ) => boolean;
+}
+
+declare interface ProcessorConstructabl0<T> {
+    new(): T;
+}
+
+declare interface ProcessorConstructabl1<T> {
     new(samplerate: number): T;
 }
 
+declare interface ProcessorConstructabl2<T> {
+    new(samplerate: number, maximumSamplerate: number): T;
+}
+
+declare interface ProcessorConstructabl4<T> {
+    new(maximumDelayMs: number, maximumSamplerate: number, maximumFrames: number, samplerate: number): T;
+}
+
```

Comment:
The effects have 4 different constructor signatures. I didn't know how to better templatize these. I'll rename them to be `ProcessorConstructable` and not `ProcessorConstructabl`